src/editor/components/modals/file-modal.jsx

```diff
@@ -0,0 +1,472 @@
+                      {
+                        // This is something like a switch statement. Only one
+                        // will render.
+                        {
+                          uploading: (
+                            <React.Fragment>
+                              <FileName>{fileName}</FileName>
+                              <FileCircularProgress
+                                variant="indeterminate"
+                                size={circularProgressSize}
+                              />
+                            </React.Fragment>
+                          ),
+                          error: (
+                            <React.Fragment>
+                              <FileName>{fileName}</FileName>
+                              <span>{fileMeta.errorMessage}</span>
+                            </React.Fragment>
+                          ),
+                          saved: (
+                            <React.Fragment>
+                              <FileNameLink href={fileURL} target="_blank">
+                                {fileName}
+                              </FileNameLink>
+                              <FilePath
+                                className="file-path"
+                                value={fileURL}
+                                readOnly
+                              />
+                              <CopiedNoticeAndFileActions>
+                                <CopiedNotice
+                                  className={
+                                    this.state.pathJustCopied ===
+                                    fileMeta.file.id
+                                      ? "visible"
+                                      : null
+                                  }
+                                >
+                                  Copied!
+                                  <Check />
+                                </CopiedNotice>
+                                <FileActions>
+                                  <FileAction
+                                    className="file-action file-action-copy"
+                                    onClick={() =>
+                                      this.copyFilePath(fileMeta.file.id)
+                                    }
+                                  >
+                                    copy path
+                                  </FileAction>
+                                  <FileAction
+                                    className="file-action file-action-delete"
+                                    onClick={() => {
+                                      this.showDeleteConfirmationModal(
+                                        key,
+                                        fileMeta.file.id,
+                                        fileName
+                                      );
+                                    }}
+                                  >
+                                    delete
```

Comment:
Agreed. I played around with fading the `FileItem` out, but it ended up being trickier than expected because CSS transitions cannot be applied to elements that are removed from the DOM. I'll take another look at this. My latest refactor, which gave each file a status, may make this easier.